concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -3509,6 +3509,87 @@ doUpdateMissedRounds pbs rds = do
             (Map.toList rds)
     storePBS pbs bsp'
 
+-- | Prime validators for suspension. Returns the subset of the given validator
+--  ids whose missed rounds exceeded the given threshold and are now primed for
+--  suspension.
+doPrimeForSuspension ::
+    ( PVSupportsDelegation pv,
+      SupportsPersistentState pv m
+    ) =>
+    PersistentBlockState pv ->
+    Word64 ->
+    [BakerId] ->
+    m ([BakerId], PersistentBlockState pv)
+doPrimeForSuspension pbs threshold bids = do
```

Comment:
Why only suspend those bakers given in `bids` and not all bakers for which `missedRounds > threshold`?

Reply:
@eb-concordium We only want to prime validators of the current payday.

Reply:
The other variant would be to remove both arguments and just look them up from the block state.

Reply:
I see 👍

Reply:
@eb-concordium I researched a little and we could look up both arguments in the `bsoPrimeValidators` computation itself. It would just mean that we end up doing unnecessary work, because both arguments are already available at the call site. Not sure what's better, a simpler API or a faster algorithm.

Reply:
Turns out that `bsoGetBakerPoolRewardDetails` also returns the current validators in the keys. So I dropped them from the arguments and kept the threshold.